Examples/Scripts/Python/vertex_fitting.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+from pathlib import Path
+from typing import Optional
+import enum
+import warnings
+
+from acts.examples import (
+    Sequencer,
+    ParticleSelector,
+    ParticleSmearing,
+    TruthVertexFinder,
+    VertexFitterAlgorithm,
+    IterativeVertexFinderAlgorithm,
+    RootParticleReader,
+    AdaptiveMultiVertexFinderAlgorithm,
+    RootVertexPerformanceWriter,
+    RootTrajectorySummaryReader,
+    TrackSelector,
+    GenericDetector,
+)
+
+import acts
+
+from acts import UnitConstants as u
+
+from common import addPythia8
+
+
+class VertexFinder(enum.Enum):
+    Truth = (1,)
+    AMVF = (2,)
+    Iterative = (3,)
+
+
+def runVertexFitting(
+    field,
+    outputDir: Path,
+    outputRoot: bool = True,
+    inputParticlePath: Optional[Path] = None,
+    inputTrackSummary: Path = None,
+    vertexFinder: VertexFinder = VertexFinder.Truth,
+    s=None,
+):
+    s = s or Sequencer(events=100, numThreads=1)
+
+    logger = acts.logging.getLogger("VertexFittingExample")
+
+    rnd = acts.examples.RandomNumbers(seed=42)
+
+    if inputParticlePath is None:
+        logger.info("Generating particles using Pythia8")
+        evGen = addPythia8(s, rnd)
+        inputParticles = evGen.config.outputParticles
+    else:
+        logger.info("Reading particles from %s", inputParticlePath.resolve())
+        assert inputParticlePath.exists()
+        inputParticles = "particles_read"
+        s.addReader(
+            RootParticleReader(
+                level=acts.logging.INFO,
+                filePath=str(inputParticlePath.resolve()),
+                particleCollection=inputParticles,
+                orderedEvents=True,
+            )
+        )
+
+    selectedParticles = "particles_selected"
+    ptclSelector = ParticleSelector(
+        level=acts.logging.INFO,
+        inputParticles=inputParticles,
+        outputParticles=selectedParticles,
+        removeNeutral=True,
+        absEtaMax=2.5,
+        rhoMax=4.0 * u.mm,
+        ptMin=500 * u.MeV,
+    )
+    s.addAlgorithm(ptclSelector)
+
+    trackParameters = "trackparameters"
+    if inputTrackSummary is None:
```

Comment:
Maybe this should be `if inputTrackSummary is None or inputParticlePath is None:` because having a inputTrackSummary doesn't mak sense if you have no input particle right ?

Reply:
I changed this.